roles/edpm_fips/tasks/enable.yml

```diff
@@ -0,0 +1,75 @@
+---
+# Copyright 2023 Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Enable FIPS
+  become: true
+  ansible.builtin.command: fips-mode-setup --enable
+  register: _fms_enable
+  changed_when: _fms_enable.rc == 0
+
+- name: Ensure that `fips=1` flag is present in GRUB parameter file
+  block:
+    - name: Check if GRUB_CMDLINE_LINUX_DEFAULT var exists
+      ansible.builtin.lineinfile:
+        path: "{{ edpm_fips_grub_params_file }}"
+        line: 'GRUB_CMDLINE_LINUX_DEFAULT="fips=1"'
+        regexp: '^GRUB_CMDLINE_LINUX_DEFAULT\=".*"$'
+      check_mode: true
+      register: _grub_params_check
+    - name: Define GRUB_CMDLINE_LINUX_DEFAULT var if doesn't exist
+      become: true
+      ansible.builtin.lineinfile:
+        path: "{{ edpm_fips_grub_params_file }}"
+        line: 'GRUB_CMDLINE_LINUX_DEFAULT="fips=1"'
+        state: present
+      when:
+        - _grub_params_check.changed
+        - _grub_params_check.msg == "line added"
+    - name: Add fips=1 parameter to GRUB_CMDLINE_LINUX_DEFAULT var
+      become: true
+      ansible.builtin.lineinfile:
+        path: "{{ edpm_fips_grub_params_file }}"
+        line: 'GRUB_CMDLINE_LINUX_DEFAULT="\1 fips=1"'
+        regexp: '^GRUB_CMDLINE_LINUX_DEFAULT\=\"(.*)\"$'
+        state: present
+        backrefs: true
+      when:
+        - _grub_params_check.changed
+        - _grub_params_check.msg == "line replaced"
+    - name: Regenerate grub configuration file
+      become: true
+      ansible.builtin.command: "grub2-mkconfig -o {{ edpm_fips_grub_config_file }}"
+      register: _g2_mkconfig
+      changed_when: _g2_mkconfig.rc == 0
+
+- name: Change iscsid CHAP Algorithms
+  vars:
+    edpm_iscsid_chap_algs: "{{ edpm_fips_iscsid_chap_algorithms }}"
+  ansible.builtin.include_role:
+    name: osp.edpm.edpm_iscsid
+    tasks_from: configure.yml
+
+- name: Reboot the node and check FIPS status
+  when: edpm_fips_reboot
+  block:
+    - name: Reboot the machine
+      ansible.builtin.reboot:
```

Comment:
could we tie this into using edpm_reboot? As long as this service comes before edpm_reboot in the dataplane services list, then all you would need to do is touch /var/lib/openstack/reboot_required. That may eliminate extra reboots.

I'm not sure how you would accomplish the following check and asserts after the reboot though, unless it was a separate dataplane service for that.